ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/storage/flat/FlatDbReaderStrategy.java

```diff
@@ -158,7 +158,7 @@ public Map<Bytes32, Bytes> streamStorageFlatDatabase(
             .streamFromKey(
                 ACCOUNT_STORAGE_STORAGE,
                 Bytes.concatenate(accountHash, startKeyHash).toArrayUnsafe())
-            .takeWhile(pair -> Bytes.wrap(pair.getKey()).slice(0, Hash.SIZE).equals(accountHash))
+            .filter(pair -> Bytes.wrap(pair.getKey()).slice(0, Hash.SIZE).equals(accountHash))
```

Comment:
I think the right answer is to add the filter above the takeWhile rather than replace with it.  That won't impact performance and it will ensure that unsorted flatDb storage doesn't prematurely end the stream (like in the case of layered)

Reply:
I ended up creating streamFromKey(start, end), that way each storage impl can do what is best for its underlying data.  Also I noticed the snapshot streamFromKey was not efficient and switched to an iterate.seek() impl